corehq/apps/dump_reload/sql/serialization.py

```diff
@@ -3,19 +3,53 @@
 
 from django.core.serializers.json import Serializer as JsonSerializer
 
+from corehq.motech.const import PASSWORD_PLACEHOLDER
 from corehq.util.json import CommCareJSONEncoder
 
 
 class JsonLinesSerializer(JsonSerializer):
     """
     Convert a queryset to JSON outputting one object per line
     """
+
+    def serialize(
+        self,
+        queryset,
+        *,
+        stream=None,
+        fields=None,
+        use_natural_foreign_keys=False,
+        use_natural_primary_keys=False,
+        progress_output=None,
+        object_count=0,
+        reset_encrypted_fields=True,
+        **options
+    ):
+        self.reset_encrypted_fields = reset_encrypted_fields
+        super().serialize(
+            queryset,
+            stream=stream,
+            fields=fields,
+            use_natural_foreign_keys=use_natural_foreign_keys,
+            use_natural_primary_keys=use_natural_primary_keys,
+            progress_output=progress_output,
+            object_count=object_count,
+            **options
+        )
+
     def start_serialization(self):
         self._init_options()
 
     def end_serialization(self):
         pass
 
+    def get_dump_object(self, obj):
+        dumped_obj = super().get_dump_object(obj)
+        if self.reset_encrypted_fields and hasattr(obj, 'encrypted_fields'):
+            for field in obj.encrypted_fields():
+                dumped_obj['fields'][field] = PASSWORD_PLACEHOLDER
```

Comment:
Use of `PASSWORD_PLACEHOLDER` seems like it's breaking an abstraction boundary. Maybe replace the `encrypted_fields` method with a dict property, allowing the implementing class to specify the appropriate value for each field?
```py
override_field_values = {
    'password': PASSWORD_PLACEHOLDER,
    'client_secret': PASSWORD_PLACEHOLDER,
}
```

Reply:
Done as part of 0762b0dfb072fc938b2d6d11445ae45b5c315ede